app/models/course.rb

```diff
@@ -185,6 +185,9 @@ def new_draft_attributes
   scope :with_study_modes, ->(study_modes) do
     where(study_mode: Array(study_modes) << "full_time_or_part_time")
   end
+  scope :with_subjects, ->(subject_codes) do
+    includes(:subjects).where(subject: { subject_code: subject_codes })
```

Comment:
Can we move the subject bit into a scope on `Subject`? That would keep the 'what is a subject code' concern in `Subject`. I think this should also be `joins` rather than `includes`.

Reply:
Yup no prob

Reply:
https://github.com/DFE-Digital/teacher-training-api/pull/1186/commits/717c08cd04b479687dc6e2a04a226afb94f7f345